backend/dataall/cdkproxy/stacks/notebook.py

```diff
@@ -62,26 +63,47 @@ def __init__(self, scope, id: str, target_uri: str = None, **kwargs) -> None:
 
         env_group = self.get_env_group(notebook)
 
+        cdk_exec_role = SessionHelper.get_cdk_exec_role_arn(notebook.AWSAccountId, notebook.region)
+
```

Comment:
Couldn't we use environment.CDKRoleArn and avoid this function?

Reply:
Yes we could also do it that way - currently notebooks stack has no helper function to get the environment, only to get the env group. So either it would be adding that helper function in the Notebook Stack to get the environment and the `environment.CDKRoleArn` property or using the `get_cdk_exec_role_arn()` defined above

This is the only use of `get_cdk_exec_role_arn()` but in future it could be used more if needed - similar to how we have `get_cdk_look_up_role_arn()`. The value is rather static where the IAM role ARN would only change if we had custom synthesizers in CDK changing the qualifier - which I see little risk for. What are your thoughts?

Reply:
2 things. 
- Independently of how we get the role arn, we need to be consistent across modules. Either we use the environment property or the method but I would not have a mix of both.
- I am more inclined to use a method exactly because of what you pointed out, custom qualifiers in CDK. Let's say that one customer wants to implement something like this. It would be relatively easy to add an SSM parameter with the qualifier and then get it in the `get_cdk_exec_role_arn()` method. Plus, we can get rid of an unnecessary column in RDS